File: src/constants/user_preferences.py
# Standard library
from typing import Any

# App enums
from app_enums import UserPreferencesKeys

USER_PREFERENCES_FILE = "user_preferences"

DEFAULT_USER_PREFERENCES: dict[UserPreferencesKeys, Any] = {
    UserPreferencesKeys.THEME: "light",
    UserPreferencesKeys.AUTOMATIC_NOTIFICATIONS: True,
    UserPreferencesKeys.LANGUAGE: "English",
    UserPreferencesKeys.LATEST_GITHUB_VERSION: "0.0.0",
    UserPreferencesKeys.DISCLAIMER_SHOWN: False,
    UserPreferencesKeys.LAST_UPDATE_CHECK: "1970-01-01",
    UserPreferencesKeys.DOWNLOAD_DIRECTORY: "",
    UserPreferencesKeys.UPGRADE_AVAILABLE: False,
    UserPreferencesKeys.WHATS_NEW_SHOWN: False,
}

File: lib/constants/user_preferences.dart
import '../enums/user_preferences.dart' show UserPreferencesKeys;

const Map<UserPreferencesKeys, dynamic> defaultUserPreferences = {
  UserPreferencesKeys.theme: 'light',
  UserPreferencesKeys.automaticNotifications: true,
  UserPreferencesKeys.language: 'en',
  UserPreferencesKeys.latestGithubVersion: '0.0.0',
  UserPreferencesKeys.disclaimerShown: false,
  UserPreferencesKeys.lastUpdateCheck: '1970-01-01',
  UserPreferencesKeys.downloadDirectory: '',
  UserPreferencesKeys.upgradeAvailable: false,
  UserPreferencesKeys.whatsNewShown: false,
};
